### PR TITLE
Fixed transaction promise mutation issue

### LINKED ISSUE #1052: Transaction promise is mutating after being resolved (resolves to undefined after a while)
Detail description https://github.com/petkaantonov/bluebird/issues/863

### PR DESCRIPTION
**Pull request by elhigu**
Fixes #1052 and fixed + disabled test case of #1970.

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -254,10 +254,10 @@ const promiseInterface = [
...
 ]
 
-// Creates a method which "coerces" to a promise, by calling a
-// "then" method on the current `Target`.
+// Creates methods which proxy promise interface methods to 
+// internal transaction resolution promise
 promiseInterface.forEach(function(method) {
   Transaction.prototype[method] = function() {
-    return (this._promise = this._promise[method].apply(this._promise, arguments))
+    return this._promise[method].apply(this._promise, arguments)
```
Comment (elhigu) [inline src/transaction.js]: This is the actual fix

File: test/integration/builder/transaction.js [hunk 2]
```diff
@@ -269,14 +269,13 @@ module.exports = function(knex) {
...
         expect(queryEventTriggered).to.equal(true);
-        done();
       }
 
-      knex.transaction(function(trx) {
-        trx.select('*').from('accounts').then(trx.commit).catch(trx.rollback);
-      })
-          .then(expectQueryEventToHaveBeenTriggered)
-          .catch(expectQueryEventToHaveBeenTriggered);
+      return knex.transaction(function(trx) {
```
Comment (elhigu) [inline test/integration/builder/transaction.js]: changed done() callback async test to just return promise

File: test/integration/builder/transaction.js [hunk 3]
```diff
@@ -398,18 +397,37 @@ module.exports = function(knex) {
...
-      })
-      .then(function() {
-        expect(true).to.equal(false, 'Transaction should not have commited');
-      })
-      .catch(function(error) {
-        expect(error instanceof Error).to.equal(true);
-        expect(error.message).to.equal('Transaction rejected with non-error: undefined');
-      });
+    it.skip('Rollback without an error should not reject with undefined #1966', function() {
+      return knex.transaction(function(tr) {
```
Comment (elhigu) [inline test/integration/builder/transaction.js]: Added missing return + disabled the test because the implemented feature is broken

**Comment by elhigu** [comment]
@wubzz could you check this out please.

**Comment by wubzz** [comment]
@elhigu I don't understand what you mean by [mysql and maria overriding](https://github.com/tgriesser/knex/commit/b722753ea3c9984d016a1419adbc39db7cee5ef1#commitcomment-214867609) `transaction.query()`, there are no dialect specific extensions of the `Transaction` class as far as I know?

**Comment by elhigu** [comment]
Check src/dialects/maria/transaction.js

**Comment by wubzz** [comment]
Ah I see it now. The copy-pasta for a single extra error handler hurts my eyes. I was under the impression transaction was unified across all dialets. Will look into this..

**Pull request merged**